drivers/macvlan/macvlan.go

```diff
@@ -51,17 +58,44 @@ type network struct {
 	endpoints endpointTable
 	driver    *driver
 	config    *configuration
+	once      *sync.Once
 	sync.Mutex
+	dbExists bool
+	dbIndex  uint64
 }
 
-// Init initializes and registers the libnetwork macvlan driver
+// Init registers a new instance of the macvlan driver
 func Init(dc driverapi.DriverCallback, config map[string]interface{}) error {
-	c := driverapi.Capability{
-		DataScope: datastore.LocalScope,
-	}
 	d := &driver{
 		networks: networkTable{},
 	}
+	// register the driver as globally scoped if a global scope label is passed
+	if labelData, ok := config[driverPrefix]; ok {
+		if labelData == globalScope {
```

Comment:
I think we should flip the defaults. Make it GlobalScoped by default. But if the user wants, they can make it a local scoped driver on a per-node basis. That is more natural since local scoped drivers means that particular node is a pet.